includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -527,6 +527,19 @@ public function init_hooks() {
 		$this->maybe_init_subscriptions_hooks();
 	}
 
+	/**
+	 * Updates icon and title using the account country.
+	 * This method runs on init is not in the controller because get_account_country might
+	 * make a request to the API if the account data is not cached.
+	 *
+	 * @return void
+	 */
+	public function update_properties_with_country(): void {
+		$account_country = $this->get_account_country();
+		$this->icon      = $this->payment_method->get_icon( $account_country );
+		$this->title     = $this->payment_method->get_title( $account_country );
```

Comment:
Would reducing the number of db cache fetches makes sense by avoiding running this logic for any payment method other than Afterpay/Clearpay? The downside of this approach is payment-method-specific implementation added to the gateway class (there are other occurrences of similar approach within the class already). Unfortunately, this comes with the fact that gateway does too much (too much includes also keeping states for `icon` & `title` of particular payment method & thus handling their updates, which ideally a payment method class should be responsible for).

My main motivation for proposing this is I'd like to avoid a store, which doesn't have Afterpay enabled, unnecessarily fetch the account country from the db cache.

```suggestion
		if ( $this->stripe_id === Payment_Method::AFTERPAY ) {
			$account_country = $this->get_account_country();
			$this->icon      = $this->payment_method->get_icon( $account_country );
			$this->title     = $this->payment_method->get_title( $account_country );
		}
```

Reply:
Great idea! I updated the code.